### ISSUE TITLE
fails to sync a folder/library

### ISSUE THREAD

**Issue by dportabella** [haiwen/seafile#1659]
My `Personal` library has 80Gb and it is already synced between my server and my laptop1.
Now I ask Seafile client to sync this library to my laptop2. It downloads 60Gb and then it fails to continue. I've tried to uninstall Seafile client and all delete all my local Seafile data and restart again, but I get the same problem (it syncs 60Gb and then it fails to continue).

laptop1 is OSX 10.10.5 and Seafile 5.0.6.
laptop2 is OSX 10.11.5 and Seafile 5.0.7.
This is a copy of seafile.log.

```
[05/24/16 23:42:39] seaf-daemon.c(542): starting seafile client 5.0.7
[05/24/16 23:42:39] seaf-daemon.c(544): seafile source code version 20ba95c458fc9586333dbe417ecfa07f5bb0afb0
[05/24/16 23:42:39] ../common/mq-mgr.c(60): [mq client] mq cilent is started
[05/24/16 23:42:39] ../common/mq-mgr.c(106): [mq mgr] publish to hearbeat mq: seafile.heartbeat

[05/24/16 23:42:39] clone-mgr.c(819): Transition clone state for 650b8294 from [init] to [connect].
[05/24/16 23:42:40] clone-mgr.c(819): Transition clone state for 650b8294 from [connect] to [fetch].
[05/24/16 23:42:40] http-tx-mgr.c(4130): Download with HTTP sync protocol version 1.
[05/24/16 23:42:40] http-tx-mgr.c(1012): Transfer repo '650b8294': ('normal', 'init') --> ('normal', 'check')
[05/24/16 23:42:40] http-tx-mgr.c(1012): Transfer repo '650b8294': ('normal', 'check') --> ('normal', 'commit')
[05/24/16 23:42:40] http-tx-mgr.c(1012): Transfer repo '650b8294': ('normal', 'commit') --> ('normal', 'fs')
[05/24/16 23:43:01] http-tx-mgr.c(1012): Transfer repo '650b8294': ('normal', 'fs') --> ('normal', 'data')

[05/24/16 23:47:32] http-tx-mgr.c(3929): Bad response code for GET https://seafile.example.com/seafhttp/repo/650b8294-05a0-4771-917f-7b14d5575eaa/block/2159c86e00f47bb367531c4a41bef1c014077bd0: 500.
[05/24/16 23:47:32] repo-mgr.c(4947): Transfer failed.
[05/24/16 23:47:32] http-tx-mgr.c(3929): Bad response code for GET https://seafile.example.com/seafhttp/repo/650b8294-05a0-4771-917f-7b14d5575eaa/block/8c14bf63376a80b134a06aa1a996492af59b5695: 500.
[05/24/16 23:47:32] repo-mgr.c(4947): Transfer failed.
[05/24/16 23:47:32] http-tx-mgr.c(3929): Bad response code for GET https://seafile.example.com/seafhttp/repo/650b8294-05a0-4771-917f-7b14d5575eaa/block/1cae59ae1a0ed05d7cac8ef6ba89946a789710d0: 500.
[05/24/16 23:47:32] repo-mgr.c(4947): Transfer failed.

[05/24/16 23:48:22] http-tx-mgr.c(707): libcurl failed to GET https://seafile.example.com/seafhttp/repo/650b8294-05a0-4771-917f-7b14d5575eaa/block/c4001fd3f2921f3e9eb6136b7f0752eb109c7143: Failed writing received data to disk/application.
[05/24/16 23:48:22] repo-mgr.c(4947): Transfer failed.
[05/24/16 23:48:22] http-tx-mgr.c(707): libcurl failed to GET https://seafile.example.com/seafhttp/repo/650b8294-05a0-4771-917f-7b14d5575eaa/block/7cff91399c514da351dbda49bf90b12955f953d9: Failed writing received data to disk/application.
[05/24/16 23:48:22] repo-mgr.c(4947): Transfer failed.
[05/24/16 23:48:22] http-tx-mgr.c(707): libcurl failed to GET https://seafile.example.com/seafhttp/repo/650b8294-05a0-4771-917f-7b14d5575eaa/block/a9d71c15c0e02675466ccb15c641c8c34d3b0d97: Failed writing received data to disk/application.
[05/24/16 23:48:22] repo-mgr.c(4947): Transfer failed.

[05/24/16 23:48:22] http-tx-mgr.c(1012): Transfer repo '650b8294': ('normal', 'data') --> ('error', 'finished')
```

---- and then it keeps repeating this same log ---
```
[05/24/16 23:48:22] clone-mgr.c(819): Transition clone state for 650b8294 from [fetch] to [fetch].
[05/24/16 23:48:22] http-tx-mgr.c(4130): Download with HTTP sync protocol version 1.
[05/24/16 23:48:22] http-tx-mgr.c(1012): Transfer repo '650b8294': ('normal', 'init') --> ('normal', 'check')
[05/24/16 23:48:22] http-tx-mgr.c(1012): Transfer repo '650b8294': ('normal', 'check') --> ('normal', 'commit')
[05/24/16 23:48:22] http-tx-mgr.c(1012): Transfer repo '650b8294': ('normal', 'commit') --> ('normal', 'fs')
[05/24/16 23:48:45] http-tx-mgr.c(1012): Transfer repo '650b8294': ('normal', 'fs') --> ('normal', 'data')

[05/24/16 23:54:01] http-tx-mgr.c(3929): Bad response code for GET https://seafile.example.com/seafhttp/repo/650b8294-05a0-4771-917f-7b14d5575eaa/block/8c14bf63376a80b134a06aa1a996492af59b5695: 500.
[05/24/16 23:54:01] repo-mgr.c(4947): Transfer failed.
[05/24/16 23:54:01] http-tx-mgr.c(3929): Bad response code for GET https://seafile.example.com/seafhttp/repo/650b8294-05a0-4771-917f-7b14d5575eaa/block/2159c86e00f47bb367531c4a41bef1c014077bd0: 500.
[05/24/16 23:54:01] repo-mgr.c(4947): Transfer failed.
[05/24/16 23:54:01] http-tx-mgr.c(3929): Bad response code for GET https://seafile.example.com/seafhttp/repo/650b8294-05a0-4771-917f-7b14d5575eaa/block/1cae59ae1a0ed05d7cac8ef6ba89946a789710d0: 500.
[05/24/16 23:54:01] repo-mgr.c(4947): Transfer failed.

[05/24/16 23:54:52] http-tx-mgr.c(707): libcurl failed to GET https://seafile.example.com/seafhttp/repo/650b8294-05a0-4771-917f-7b14d5575eaa/block/7cff91399c514da351dbda49bf90b12955f953d9: Failed writing received data to disk/application.
[05/24/16 23:54:52] repo-mgr.c(4947): Transfer failed.
[05/24/16 23:54:52] http-tx-mgr.c(707): libcurl failed to GET https://seafile.example.com/seafhttp/repo/650b8294-05a0-4771-917f-7b14d5575eaa/block/ab7af02daff5a67ad189af5bd01c8494edf19e58: Failed writing received data to disk/application.
[05/24/16 23:54:52] repo-mgr.c(4947): Transfer failed.
[05/24/16 23:54:52] http-tx-mgr.c(707): libcurl failed to GET https://seafile.example.com/seafhttp/repo/650b8294-05a0-4771-917f-7b14d5575eaa/block/c195bf8cbf040dfb37658e9eacacd939770a2436: Failed writing received data to disk/application.
[05/24/16 23:54:52] repo-mgr.c(4947): Transfer failed.

[05/24/16 23:54:52] http-tx-mgr.c(1012): Transfer repo '650b8294': ('normal', 'data') --> ('error', 'finished')

```

**Comment by killing** [comment]
There should be some error on the server. The log reports 500 internal server error. Can you check the log on the server?

**Comment by dportabella** [comment]
this is what we get in the seafile server:
`other_vhosts_access.log`:
```
[02/Jun/2016:17:26:27 +0000] "GET /seafhttp/repo/650b8294-05a0-4771-917f-7b14d5575eaa/block/1cae59ae1a0ed05d7cac8ef6ba89946a789710d0 HTTP/1.1" 500 315 "-" "Seafile/5.1.1 (Apple OS X)"

[02/Jun/2016:17:26:26 +0000] "GET /seafhttp/repo/650b8294-05a0-4771-917f-7b14d5575eaa/block/8c14bf63376a80b134a06aa1a996492af59b5695 HTTP/1.1" 500 1702 "-" "Seafile/5.1.1 (Apple OS X)"
```

`seafile.log`:
`[06/02/2016 05:26:27 PM] http-server.c(1342): Failed to allocate 0 bytes memeory.
`

**Comment by dportabella** [comment]
more info: my Personal folder contained four files with zero bytes. Once I removed them from the server, then the client started to work again and sync.

**Comment by killing** [comment]
What's your server version?

**Comment by dportabella** [comment]
the last one, 5.1.3 64bit.

**Comment by killing** [comment]
Do you remember from which client you uploaded the four zero size files? Usually zero size files should have ID all-0 and won't be uploaded/downloaded. It seems their IDs don't match their contents.

**Comment by dportabella** [comment]
the server administrator says that these four zero-byte files didn't have a name, and that using `seaf-fsck` might have solved the issue.
I don't have more info about this, so i guess it's best to close this issue.

**Comment by cunhaal** [comment]
I had the same issue. 
I´m running seafile on a raspberry pi. The client on my android would work fine. The cliend on windows 10 would sync and give server error. 
seafile.log displayed hundreds of times this line:
[04/20/17 11:48:45] http-server.c(1374): Failed to allocate 0 bytes memeory.

After doing searching for a solution everywhere, I also discovered my library was corruped and the problem was solved by ./seaf-fsck --repair.
